crates/ursa/src/ursa/mod.rs

```diff
@@ -55,38 +54,22 @@ pub struct CliOpts {
 
 impl CliOpts {
     pub fn to_config(&self) -> Result<UrsaConfig> {
-        let mut path = PathBuf::from(env!("HOME")).join(DEFAULT_CONFIG_PATH_STR);
-        if let Some(config_file) = &self.config {
-            info!(
-                "Reading configuration from user provided config file {}",
-                config_file
-            );
-            path = PathBuf::from(&config_file);
+        let mut config = load_config(
+            &self
+                .config
+                .as_ref()
+                .map(|p| PathBuf::from(p).resolve().to_path_buf())
+                .unwrap_or_else(|| PathBuf::from(env!("HOME")).join(DEFAULT_CONFIG_PATH_STR)),
```

Comment:
`env!` macro is executed at compile time:

> Inspects an environment variable at compile time [[source](https://doc.rust-lang.org/std/macro.env.html)]

so this is not portable code and the generated binary will break on another machine.

Reply:
https://github.com/fleek-network/ursa/pull/116/commits/7c838e19cbc3163ba3c4fce109b4f510b29135b6